Stop filters from being anchored to the beginning & filter help page improvements

wylfen opened the pull request:
See #1073.

yagebu commented on fava/help/filters.md:
```diff
@@ -30,14 +30,13 @@ the account name, or a regular expression matching the account name, e.g.
--   Search payee and narration by supplying a simple string, which again is
-    matched as a regular expression. For Note directives, the comment will be
-    searched.
+-   Filter by any entry attribute, such as payee `payee:".*restaurant.*"` or
```
The `.*`s can be removed. Maybe you could add an example with an anchored regex?

wylfen commented:
Removed the `.*` from all examples where they could be dropped or replaced with anchors. Also noticed that the help page still referred to `statement:`, which I changed to `document:`.

Also now has examples with both `$` and `^`.

Rebased on current master.
